app/frontend/src/service/client.ts

```diff
@@ -37,12 +38,23 @@ export class AuthInterceptor {
   }
 }
 
-const interceptor = new AuthInterceptor();
+class TimeoutInterceptor {
+  async intercept(request: any, invoker: (request: any) => any) {
+    const deadline = Date.now() + grpcTimeout;
+    const metadata = request.getMetadata();
+    metadata.deadline = deadline;
+    const response = await invoker(request);
```

Comment:
In theory this works but I was only able to get a "Deadline exceeded" error once, even setting `grpcTimeout` to one and limiting my bandwidth! 🤔

Reply:
What happens if this is set to 0?

Reply:
What happens if this is set to 0?

Reply:
Still nothing... I can see there is correctly a  `grpc-timeout: 10000m` header on the request.